modules/widget-visibility/widget-conditions.php

```diff
@@ -235,6 +235,50 @@ public static function widget_conditions_options_echo( $major = '', $minor = ''
 				</optgroup>
 				<?php
 			break;
+
+			case 'single_post':
+				// Get Post object.
+				$post_types = array(
+					'post' => get_post_type_object( 'post' )
+				);
+
+				$custom_post_types = get_post_types(
+					array(
+						'public'   => true,
+						'_builtin' => false,
+					),
+					'objects'
+				);
+
+				// Add "Post" object to other post types.
+				$post_types = array_merge( $post_types, $custom_post_types );
+
+				foreach ( $post_types as $post_type ) {
+				?>
+					<optgroup label="<?php echo esc_attr_x( $post_type->label, 'jetpack' ); ?>">
```

Comment:
That won't work either I"m afraid. You can't pass variables to translation functions in general. This is explained a bit here:
https://codex.wordpress.org/I18n_for_WordPress_Developers#Placeholders